cocotb/__init__.py

```diff
@@ -192,6 +192,29 @@ def _initialise_testbench(argv_):
     if not sys.warnoptions:
         warnings.simplefilter("default")
 
+    attach_str = os.getenv("COCOTB_PY_ATTACH")
+    if attach_str is not None:
+        try:
+            if ":" not in attach_str:
+                raise ValueError("Invalid COCOTB_PY_ATTACH, form is 'host:port'.")
+            host_str, port_str = attach_str.split(":", 1)
+            port_num = int(port_str)
+        except Exception as e:
+            raise RuntimeError("Failure to parse COCOTB_PY_ATTACH ('{}')".format(attach_str)) from e
+
+        try:
+            import debugpy
+        except ImportError as e:
+            raise RuntimeError("COCOTB_PY_ATTACH is set but debugpy package is not importable. Install debugpy to debug cocotb.") from e
+
+        try:
+            listen_host, listen_port = debugpy.listen((host_str, port_num))
+            cocotb.log.info("Waiting for Python debugger attach on {}:{}".format(listen_host, listen_port))
+            debugpy.wait_for_client()
+            debugpy.breakpoint()
+        except Exception as e:
+            raise RuntimeError("COCOTB_PY_ATTACH failure using '{}:{}'".format(host_str, port_num)) from e
+
```

Comment:
Do you think you could factor this into another function? This startup routine is starting to become hefty...